frontend/components/course/Tile.tsx

```diff
@@ -2,17 +2,99 @@ import Image from "next/image";
 import Link from "next/link";
 import { FC } from "react";
 
+import { EnrollmentStatus_enum } from "../../__generated__/globalTypes";
+import {
+  enrollmentStatusForCourse,
+  hasEnrollments,
+  hasProgram,
+} from "../../helpers/courseHelpers";
 import { CourseList_Course } from "../../queries/__generated__/CourseList";
+import { CourseListWithEnrollments_Course } from "../../queries/__generated__/CourseListWithEnrollments";
+import { CourseWithEnrollment_Course_by_pk_Enrollments } from "../../queries/__generated__/CourseWithEnrollment";
 
 interface IProps {
-  course: CourseList_Course;
+  course: CourseList_Course | CourseListWithEnrollments_Course;
 }
 
+const colorForEnrollmentStatus = (
+  status: EnrollmentStatus_enum | "NOT_APPLIED",
+  enrollment: CourseWithEnrollment_Course_by_pk_Enrollments | undefined
+): string => {
+  if (
+    status === EnrollmentStatus_enum.APPLIED ||
+    status === EnrollmentStatus_enum.CONFIRMED
+  ) {
+    return "bg-edu-course-current";
+  }
+  if (
+    status === EnrollmentStatus_enum.REJECTED ||
+    status === EnrollmentStatus_enum.ABORTED
+  ) {
+    return "bg-gray-300";
+  }
+  if (status === EnrollmentStatus_enum.INVITED) {
+    if (enrollment && new Date() < enrollment.ExpirationDate) {
+      return "bg-edu-course-invited";
+    } else {
+      return "bg-gray-300";
```

Comment:
Having a comment here could be helpful for the future.
```suggestion
     // invitation is expired
     return "bg-gray-300";
```